frontend/packages/multicluster-sdk/src/api/useHubClusterName.ts

```diff
@@ -1,43 +1,28 @@
 /* Copyright Contributors to the Open Cluster Management project */
-import { fetchHubClusterName, getCachedHubClusterName } from '../internal/cachedHubClusterName'
 import { UseHubClusterName } from '../types'
 import { useEffect, useMemo, useState } from 'react'
-import { useIsFleetAvailable } from './useIsFleetAvailable'
+import { cachedFleetConfiguration, useFleetConfiguration } from '../internal/useFleetConfiguration'
 
 /**
  * Hook that provides hub cluster name.
  *
  * @returns Array with `hubclustername`, `loaded` and `error` values.
  */
 export const useHubClusterName: UseHubClusterName = () => {
-  const cachedHubClusterName = getCachedHubClusterName()
-  const [hubClusterName, setHubClusterName] = useState<string | undefined>(cachedHubClusterName)
-  const [loaded, setLoaded] = useState<boolean>(!!cachedHubClusterName)
-  const [error, setError] = useState<any>(undefined)
+  const [hubClusterName, setHubClusterName] = useState<string | undefined>()
 
-  const fleetAvailable = useIsFleetAvailable()
+  const [fleetConfiguration, loaded, error] = useFleetConfiguration()
 
   useEffect(() => {
-    if (!fleetAvailable) {
-      setHubClusterName(undefined)
-      setLoaded(false)
-      setError('A version of RHACM that is compatible with the multicluster SDK is not available')
-      return
-    }
-
-    const currentCachedName = getCachedHubClusterName()
-    if (!currentCachedName) {
-      void (async () => {
-        try {
-          const hubName = await fetchHubClusterName()
-          setHubClusterName(hubName)
-          setLoaded(true)
-        } catch (err) {
-          setError(err)
-        }
-      })()
-    }
-  }, [fleetAvailable])
+    setHubClusterName(fleetConfiguration?.localHubName)
+  }, [fleetConfiguration])
 
   return useMemo(() => [hubClusterName, loaded, error], [hubClusterName, loaded, error])
 }
+
+/**
+ * Get cached hub cluster name without triggering a re-fetch.
+ */
+export const getHubClusterName = (): string | undefined => {
+  return cachedFleetConfiguration?.localHubName
+}
```

Comment:
I this supposed to be API? What if no one has called it before to get the cache initialized?

Reply:
very good point--I just renamed fetchFleetObservabilityInstalled to getCachedFleetObservabilityInstalled

instead of it possibly returning null if the hook hasn't been called, it gets the same promise that the hook uses

it must await getCachedFleetObservabilityInstalled but it'll never be null and /hub won't be called multiple times